### PR TITLE
TECH-469: Update cucumber test for PUT /data/registryname/versionnumber/update

### PR DESCRIPTION
**Pull request by mdulko**
### **Description**
Update cucumber test for PUT /data/registryname/versionnumber/update
Make sure tests contain

- smoke tests
- sanity performance tests
- no hard-coded data
- header tests
- etc. that are described in https://govstack-global.atlassian.net/wiki/spaces/GH/pages/157712385/Gherkin+file+template
- API spec: https://github.com/GovStackWorkingGroup/bb-payments/blob/main/api/UpdateBeneficiary.yaml

Acceptance criteria:

- test is updated
- test match with accepted template

Ticket: https://govstack-global.atlassian.net/browse/TECH-469

### CODE REVIEW

File: test/openAPI/features/data_update.feature [hunk 1]
```diff
@@ -1,17 +1,25 @@
...
+  @positive @unit
+  Scenario: Successfully update user information record in database
+    Given User wants to update existing record in database
+    When PUT request to check if the record exists in the database is sent with given path params "registryname" as registryname and "111" as versionnumber
+    And the request contains a payload with given "EE378627348834" as ID "John" as FirstName "Helmut" as LastName and "RR-1234567889" as BirthCertificateID and the request overwrites the record with given "EE378627348834" as updatedID "John" as updatedFirstName "Helmut" as updatedLastName and "RR-1234567889" as updatedBirthCertificateID
+    Then response from /data/{registryname}/{versionnumber}/update is received
+    And the response from /data/{registryname}/{versionnumber}/update should be returned in a timely manner 15000ms
+    And the response from /data/{registryname}/{versionnumber}/update should have status 200
+    And the response from /data/{registryname}/{versionnumber}/update should have content-type: application/json header
+    And the response from /data/{registryname}/{versionnumber}/update should match json schema
```
[the remark this answers is not in the document]
Comment (mgrochalskisoldevelo) [reply test/openAPI/features/data_update.feature:24]: image

File: test/openAPI/features/data_update.feature [hunk 2]
```diff
@@ -1,17 +1,25 @@
...
-  Request endpoint: PUT /data/{registryname}/{versionnumber}/update
+@method=GET @endpoint=/data/{registryname}/{versionnumber}/update
+Feature: API endpoint that allows users to update a new record in database.
 
-  Scenario: The user successfully updates the record in the Digital Registries database
-    Given The user wants to update a record in the Digital Registries database and a record exists
-    When The user sends a valid request to update the record in the database
-    Then The operation to update a record is completed successfully
+  @smoke
+    Scenario: Successfully update user information record in database smoke type test
```
Comment (mgrochalskisoldevelo) [inline test/openAPI/features/data_update.feature:5]: smoke should also take all params required for "query" and "write" in the payload

File: test/openAPI/features/data_update.feature [hunk 3]
```diff
@@ -1,17 +1,25 @@
...
-
-  Scenario: The user is not able to update a record in the Digital Registries database because of an invalid request
-    Given The user wants to update a record in the Digital Registries database and a record exists
-    When The user sends an invalid request to update a new record in the database
-    Then The result of the operation to update a record returns an error due to an invalid request
+  @positive @unit
+  Scenario: Successfully update user information record in database
+    Given User wants to update existing record in database
+    When PUT request to check if the record exists in the database is sent with given path params "registryname" as registryname and "111" as versionnumber
+    And the request contains a payload with given "EE378627348834" as ID "John" as FirstName "Helmut" as LastName and "RR-1234567889" as BirthCertificateID and the request overwrites the record with given "EE378627348834" as updatedID "John" as updatedFirstName "Helmut" as updatedLastName and "RR-1234567889" as updatedBirthCertificateID
```
Comment (mgrochalskisoldevelo) [inline test/openAPI/features/data_update.feature:19]: Use examples table for it with let's say 3 examples

**Review by mgrochalskisoldevelo — CHANGES_REQUESTED** [review]
Also there is something wrong in the mockoon. To test it set a new response with the status code 400, after this one with status code 200. And set 400 as the default one. If tests will return 200 everything is okay, if 400 there is something to fix in the `Rules`. Remove the 400 status code response after

File: test/openAPI/features/data_update.feature [hunk 4]
```diff
@@ -1,17 +1,26 @@
...
+    And the response from /data/{registryname}/{versionnumber}/update should be returned in a timely manner 15000ms
+    And the response from /data/{registryname}/{versionnumber}/update should have status 200
 
-  Scenario: The user is not able to update a record in the Digital Registries database because of an invalid request
-    Given The user wants to update a record in the Digital Registries database and a record exists
-    When The user sends an invalid request to update a new record in the database
-    Then The result of the operation to update a record returns an error due to an invalid request
+    Examples:
+      | updatedID | updatedFirstName | updatedLastName | updatedBirthCertificateID |
+      | EE378627348833 | Jack | Black | RR-1234567880 |
```
Comment (mgrochalskisoldevelo) [inline test/openAPI/features/data_update.feature:24]: I believe that id should be the same and does not change after update

File: test/openAPI/features/data_update.feature [hunk 5]
```diff
@@ -1,17 +1,26 @@
...
-  Request endpoint: PUT /data/{registryname}/{versionnumber}/update
+@method=GET @endpoint=/data/{registryname}/{versionnumber}/update
+Feature: API endpoint that allows users to update a new record in database.
 
-  Scenario: The user successfully updates the record in the Digital Registries database
-    Given The user wants to update a record in the Digital Registries database and a record exists
-    When The user sends a valid request to update the record in the database
-    Then The operation to update a record is completed successfully
+  @smoke
+    Scenario: Successfully update user information record in database smoke type test
```
Comment (mgrochalskisoldevelo) [inline test/openAPI/features/data_update.feature:5]:
```suggestion
    Scenario: Successfully updates user information record in the database smoke type test
```

File: test/openAPI/features/data_update.feature [hunk 6]
```diff
@@ -1,17 +1,26 @@
...
+    Given User wants to update existing record in database
+    When PUT request to check if the record exists in the database is sent with given path params "registryname" as registryname and "111" as versionnumber
```
Comment (mgrochalskisoldevelo) [inline test/openAPI/features/data_update.feature:7]: you don't check if the record exists, but tries to update it

File: test/openAPI/features/support/helpers/helpers.js [hunk 7]
```diff
@@ -97,6 +97,55 @@ module.exports = {
...
+            'ID',
+            'FirstName',
+            'LastName',
+            'BirthCertificateID'
+          ]
+      }
+    },
+    required: ['content']
+  },
+  write: {
```
Comment (mgrochalskisoldevelo) [inline test/openAPI/features/support/helpers/helpers.js:126]: write is also marked as required in the api spec, please add it here

**Review by KarolinaKopacz — APPROVED** [review]

File: test/openAPI/features/data_update.feature [hunk 8]
```diff
@@ -1,17 +1,26 @@
...
 
-  Scenario: The user cannot update the record because the record does not exist in the Digital Registries database
-    Given The user wants to update the record in the Digital Registries database and the record does not exist
-    When The user sends a valid request to update the record in the database
-    Then The result of the operation to update the record is an error because the record does not exist in the database
+  @positive @unit
+  Scenario Outline: Successfully update user information record in database
+    Given User wants to update existing record in database
+    When PUT request to check if the record exists in the database is sent with given path params "registryname" as registryname and "111" as versionnumber
+    And the request contains a payload with given "EE378627348834" as ID "John" as FirstName "Helmut" as LastName and "RR-1234567889" as BirthCertificateID and the request overwrites the record with given "<updatedID>" as updatedID "<updatedFirstName>" as updatedFirstName "<updatedLastName>" as updatedLastName and "<updatedBirthCertificateID>" as updatedBirthCertificateID
```
[the remark this answers is not in the document]
Comment (mgrochalskisoldevelo) [reply test/openAPI/features/data_update.feature:17]: in examples it can stay like it is

**Review by mgrochalskisoldevelo — CHANGES_REQUESTED** [review]

**Review by KarolinaKopacz — APPROVED** [review]

**Review by ingmarvali — APPROVED** [review]

**Pull request merged**